src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -19,6 +19,20 @@
 #include <Kokkos_Sort.hpp> // min_max_functor
 #include <Kokkos_View.hpp>
 
+#if defined(KOKKOS_ENABLE_CUDA)
+#if (KOKKOS_COMPILER_CLANG < 900)
+// Clang of version less than 9.0 cannot compile Thrust, failing with errors
+// like this:
+//    <snip>/thrust/system/cuda/detail/core/agent_launcher.h:557:11:
+//    error: use of undeclared identifier 'va_printf'
+// Defining _CubLog here allows us to avoid that code path, however disabling
+// some debugging diagnostics
+#define _CubLog
+#endif
+#include <thrust/device_ptr.h>
+#include <thrust/sort.h>
```

Comment:
Define the macro only if it is not already defined and `#undef` it after the header includes if you defined it.

Reply:
No to the first. If someone is using ArborX and included Thrust headers before ArborX, it will likely break the build.

Reply:
It would likely be good to save the current definition if it is defined (, redefine it) and restore it after the includes. Otherwise, users get annoying warnings